framework/src/postprocessors/ParsedPostprocessor.C

```diff
@@ -17,9 +17,12 @@ ParsedPostprocessor::validParams()
   InputParameters params = GeneralPostprocessor::validParams();
   params += FunctionParserUtils<false>::validParams();
 
+  params.addDeprecatedCustomTypeParam<std::string>(
+      "function", "FunctionExpression", "The user defined function.", "Use 'expression' instead.");
   params.addRequiredCustomTypeParam<std::string>(
```

Comment:
you can have deprecated + a required with the old system for deprecation, which is why Griffin is failing

what you have to do is use the new system for deprecation.
add the old parameter as required still, but then add a line for 
```
params.deprecateParam(....)
```

Reply:
losing you a bit on that explanation...but I think you mean to make `function` required still instead of `expression`. Wouldn't that cause the same problem if `expression` was provided but not `function`? What about doing a check in the constructor to make sure one is provided?

Reply:
Oh I see an example now in `ParsedAux`

Reply:
no on both counts.

see scalarkernels/ParsedODEKernel.C for an example on how to do this